std/algorithm/iteration.d

```diff
@@ -1034,6 +1049,96 @@ template each(alias pred = "a")
     assert(res == [9, 12, 15]);
 }
 
+// #15357: `each` should behave similar to foreach
+/// each should work with index
+@safe unittest
+{
+    import std.range : iota, lockstep;
```

Comment:
> even though it's in same module, so the unittest can be run in the online documentation.

Luckily only external dependencies (e.g. global imports), it's fairly trivial to a `import x.y.x` and we do so:

https://github.com/dlang/dlang.org/blob/master/js/run_examples.js#L29

The automatic tests extractor which is partially enabled for Phobos [does this as well](https://github.com/dlang/tools/blob/master/phobos_tests_extractor.d#L64) and will warn if someone tries to sneak public tests in that aren't runnable ;-)